blog/2021-12-29-auto-setup.md

```diff
@@ -0,0 +1,121 @@
+---
+tags:
+  - Temporal
+  - Server
+posted_on_: 2021-12-29T00:00:09Z
+slug: auto-setup
+title: How Temporal Auto-Setup Works
+author: swyx
+author_title: Head of Developer Experience
+author_image_url: https://avatars.githubusercontent.com/u/6764957?v=4
+release_version: V1.14.0
+---
+
+We often say that “Temporal Server is a simple Go binary”, and yet most users are insulated from using this binary directly. There are several layers of abstraction to get through (docker-compose or Helm Chart, Dockerfile, and the auto-setup.sh script) before you reach this baseline understanding of Temporal. Together they comprise the environment requirements of Temporal:
+
+<!--truncate-->
+
+![image](https://user-images.githubusercontent.com/6764957/147678999-883be1b4-4d32-4c89-84a4-00e8b701cdef.png)
+
+While the other pieces are industry standard formats with explanations available elsewhere, the Auto-Setup is the most proprietary of them, and so essential that we often recommend users just use the [temporalio/auto-setup](https://hub.docker.com/r/temporalio/auto-setup) Docker image, over our Docker image with Temporal Server itself (most beginners don’t even know this because we abstract it with docker-compose). 
```

Comment:
```suggestion
While the other pieces are industry standard formats with explanations available elsewhere, the auto-setup is the most proprietary of them, and so essential that we often recommend users just use the [temporalio/auto-setup](https://hub.docker.com/r/temporalio/auto-setup) Docker image, over our Docker image with Temporal Server itself (most beginners don’t even know this because we abstract it with docker-compose). 
```